pkgs/applications/misc/twitch-chat-downloader/default.nix

```diff
@@ -20,6 +21,14 @@ buildPythonApplication rec {
     hash = "sha256-mV60ygrtQa9ZkJ2CImhAV59ckCJ7vJSA9cWkYE2xo1M=";
   };
 
+  patches = [
+    # Updates client_id
+    (fetchpatch {
+      url = "https://github.com/TheDrHax/Twitch-Chat-Downloader/commit/23de61a0221d3796b706774283111734dd2338e6.patch";
+      hash = "sha256-qwFRyV1MeC4Isc2gVEYStr6ytP7EzN5WuZOwQ8KOqTc=";
+    })
```

Comment:
We generally like to avoid `fetchpatch`ing unmerged PRs since GitHub can garbage collect the commits. Usually we would vendor the `.patch` file in the tree instead, but since this is so trivial you could just put `substituteInPlace tcd/example.settings.json --replace-fail old_client_id new_client_id` in a `postPatch` phase instead.

Reply:
That is surprising, I've mostly come across `fetchpatch`es of unmerged PRs, and actually had my vendored `.patch` replaced with one in another package. 
`substituteInPlace` makes a lot more sense here.

Reply:
Yeah, unfortunately the pitfalls of unmerged GitHub PRs aren‘t too widely known. I had to be convinced myself recently :)